Sources/Rendering/Core/ColorTransferFunction/index.js

```diff
@@ -430,12 +430,17 @@ function vtkColorTransferFunction(publicAPI, model) {
     return rgb[2];
   };
 
+  publicAPI.usingLogScale = () => model.scale === Scale.LOG10;
+
   //----------------------------------------------------------------------------
   // Returns a table of RGB colors at regular intervals along the function
   publicAPI.getTable = (xStart_, xEnd_, size, table) => {
+    let usingLogScale = publicAPI.usingLogScale();
     // To handle BigInt limitation
-    const xStart = Number(xStart_);
-    const xEnd = Number(xEnd_);
+    const xStart = usingLogScale
+      ? Math.log10(Number(xStart_))
+      : Number(xStart_);
+    const xEnd = usingLogScale ? Math.log10(Number(xEnd_)) : Number(xEnd_);
```

Comment:
If `usingLogScale` is set to false on [line 486](https://github.com/Kitware/vtk-js/pull/3238/files#diff-838837b0916ace672b3d0719343881d570cd87db3f95cfcaa117926495ee3e64L481), should we still set xStart and xEnd on a log scale initially? It seems like the range should be checked for validity prior to using a log scale.